docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,785 @@
+with "ALLOW" action and with a single "MeshTrafficPermission" policy. The policy has permission "any: true" and the list of "principals".
+List of principals generated based on the list from step 3. Each principal is matched with OR semantic: 
+
+```yaml
+action: ALLOW
+policies:
+  "MeshTrafficPermission":
+    permissions:
+      - any: true
+    principals: 
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/service/web"
+      - and_ids:
+          ids:
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://env/dev"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/service/web"
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://env/dev"
+            - authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/service/web"
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+            - authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/service/web"
+      - and_ids:
+          ids:
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/service/web"
+      - and_ids:
+          ids:
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/service/web"
+```
+Set this policy as "rules" into NetworkFilter.
+
+4. Shadow rules. Take only rules that have "DENY_WITH_SHADOW_ALLOW":
```

Comment:
Why don't we take `ALLOW_WITH_SHADOW_DENY`?

Reply:
We don't generate `deny` entries. Deny happens if there is no `allow` entry.